src/particles/ImpactXParticleContainer.cpp

```diff
@@ -82,4 +82,41 @@ namespace impactx
 //        Redistribute();
 
     }
+
+    std::tuple<
+            amrex::ParticleReal, amrex::ParticleReal,
+            amrex::ParticleReal, amrex::ParticleReal,
+            amrex::ParticleReal, amrex::ParticleReal>
+    ImpactXParticleContainer::MinAndMaxPositions ()
+    {
+        using PType = ImpactXParticleContainer::SuperParticleType;
+
+        amrex::ReduceOps<amrex::ReduceOpMin, amrex::ReduceOpMin, amrex::ReduceOpMin, amrex::ReduceOpMax, amrex::ReduceOpMax, amrex::ReduceOpMax> reduce_ops;
+        auto r = amrex::ParticleReduce<amrex::ReduceData<amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal>>(
+            *this,
+            [=] AMREX_GPU_DEVICE(const PType& p) noexcept -> amrex::GpuTuple<amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal, amrex::ParticleReal>
+            {
+                amrex::ParticleReal x = p.pos(0);
+                amrex::ParticleReal y = p.pos(1);
+                amrex::ParticleReal z = p.pos(2);
+                return {x, y, z, x, y, z};
+            },
+            reduce_ops);
+
+        x_min = amrex::get<0>(r);
```

Comment:
Put this directly in a vector, as shown here:
Put those directly in a vector, as shown in the Reduce link below:

https://github.com/ECP-WarpX/WarpX/blob/583c78e563f7a74e89fb49ff77c7a93c0baaad78/Source/Diagnostics/ReducedDiags/BeamRelevant.cpp#L324-L340